plugins/autocomplete/plugin.js

```diff
@@ -1344,6 +1362,20 @@
 		}, {} );
 	}
 
+	function removeLeadingSpace( editor ) {
+		var selection = editor.getSelection(),
+			nextNode = selection.getRanges()[ 0 ].getNextNode( function( node ) {
+			return Boolean( node.type == CKEDITOR.NODE_TEXT && node.getText() );
+		} );
+
+		if ( nextNode && nextNode.getText().match( /^\s+/ ) ) {
+			var range = editor.createRange();
```

Comment:
There should be only one var statement at the beginning of the function.

Reply:
`var` statement in most cases is the most readable when it's as close as possible to the first usage - as it's a subjective matter, it still objective that in the above use case you will unnecessary create a range object if `if` statement fail.

Reply:
You don't have to assign value with declaration of this variable, so range don't have to be created in such case and you just assign proper value inside `if` statement.
As it's quite small function and there is hoisting, from my perspective it's better to declare it at the beginning of scope where it used. However I understand your concerns and let remain it for main review to decide wether current approach should be changed or not.